### PR TITLE
Make map symmetrical if it's not wrapped

### PR DESCRIPTION
**Pull request by lishaoxia1985**

### CODE REVIEW

File: core/src/com/unciv/ui/utils/ZoomableScrollPane.kt [hunk 1]
```diff
@@ -11,10 +11,6 @@ open class ZoomableScrollPane: ScrollPane(null) {
     var continuousScrollingX = false
 
     init{
-        // Remove the existing inputListener
-        // which defines that mouse scroll = vertical movement
-        val zoomListener = listeners.last { it is InputListener && it !in captureListeners }
```
Comment (yairm210) [inline core/src/com/unciv/ui/utils/ZoomableScrollPane.kt:16]: I remember there's a reason this is here. Don't remember what it is, though

File: core/src/com/unciv/ui/map/TileGroupMap.kt [hunk 2]
```diff
@@ -120,9 +120,11 @@ class TileGroupMap<T: TileGroup>(tileGroups: Collection<T>, private val leftAndR
 
         // there are tiles "below the zero",
         // so we zero out the starting position of the whole board so they will be displayed as well
-        // The width has to be lowered by groupSize because wrapped maps are missing a tile on the right.
+        // Map's width is reduced by groupSize if it is wrapped, because wrapped map will miss a tile on the right.
         // This ensures that wrapped maps have a smooth transition.
-        setSize(topX - bottomX + leftAndRightPadding * 2 - groupSize, topY - bottomY + topAndBottomPadding * 2)
+        // If map is not wrapped, Map's width doesn't need to be reduce by groupSize
+        if (worldWrap) setSize(topX - bottomX + leftAndRightPadding * 2 - groupSize, topY - bottomY + topAndBottomPadding * 2)
```
Comment (yairm210) [inline core/src/com/unciv/ui/map/TileGroupMap.kt:126]: I'm pretty sure this already existed, when did this disappear?

File: core/src/com/unciv/ui/utils/ZoomableScrollPane.kt (hunk 1 above)
Comment (lishaoxia1985) [reply core/src/com/unciv/ui/utils/ZoomableScrollPane.kt:16]: compile give a warning "**Calling non-final function removeListener in constructor**"

File: core/src/com/unciv/ui/map/TileGroupMap.kt (hunk 2 above)
Comment (lishaoxia1985) [reply core/src/com/unciv/ui/map/TileGroupMap.kt:126]: This edit is in the commit #3991, but you revert it, so I split it.

**Comment by logicminimal** [comment]
what exactly does make map symetrical mean? does world wrap even need ot to function?

**Comment by lishaoxia1985** [comment]
@logicminimal Like this, if you don't use the wrap map and this commit, you will find the background map is not in the center of the screen.
YueR

**Comment by logicminimal** [comment]
> @logicminimal Like this, if you don't use the wrap map and this commit, you will find the background map is not in the center of the screen.
> YueR
> 

ok, does this change ingame world wrap though?

**Comment by lishaoxia1985** [comment]
@logicminimal No, When you play the game it is meaningless because in the game the map is not in the center of screen usually.

**Pull request merged**